lib/dialects/abstract/query.js

```diff
@@ -7,6 +7,46 @@ const Dot = require('dottie');
 const deprecations = require('../../utils/deprecations');
 const uuid = require('uuid/v4');
 
+const stringify = obj => obj instanceof Buffer ? obj.toString('hex') : obj;
+
+// Calculates the array prefix of a key (['User', 'Results'] for 'User.Results.id')
+const makeDoublePrefixMemoizer = () => {
+  const keyPrefixMemo = {};
+  return str => {
+    // We use a double memo and keyPrefixString so that different keys with the same prefix will receive the same array instead of differnet arrays with equal values
+    if (!Object.prototype.hasOwnProperty.call(keyPrefixMemo, str)) {
+      const prefixString = str.substr(0, str.lastIndexOf('.'));
+      if (!Object.prototype.hasOwnProperty.call(keyPrefixMemo, prefixString)) {
+        keyPrefixMemo[prefixString] = prefixString ? prefixString.split('.') : [];
+      }
+      keyPrefixMemo[str] = keyPrefixMemo[prefixString];
+    }
+    return keyPrefixMemo[str];
+  };
+};
+
+/**
+ * Calculate the last item in the array prefix ('Results' for 'User.Results.id')
+ *
+ * @param {string} str
+ */
+const secondSuffix = str => {
+  const lastDelim = str.lastIndexOf('.');
```

Comment:
Will this work?

```js
  if (!str.includes('.')) return '';

  const array = str.split('.');
  return array[array.length - 1];
```

Reply:
I tried this approach and it was significantly slower.